### ISSUE TITLE
Firmware write failed 

### ISSUE THREAD

**Issue by leledumbo** [Freeyourgadget/Gadgetbridge#176]
Gadget: Mi Band 1 (color LED version)
Firmware version: 1.0.11.6
Device: Mi4c
Gadgetbridge version: 0.6.6 (from f-droid)

It all begins two days ago when my Mi Band stops tracking activity / sleep data. Only the step counter works, everything else is not recorded. If I check Mi Fit at night, it always says I need <my daily steps goal, call it g> to goal despite the step counter shows that I've walked for n steps. Previously, the correct x - n is displayed.

So I decide to downgrade my firmware. Reading here and there, I found that Gadgetbridge has this ability. After some struggling to pair the band I try to flash 1.0.10.11 firmware and Gadgetbridge says that it's known and tested, so I proceed with flashing. However, no matter how many times I try, in the end it's always "Firmware write failed" (with DO NOT REBOOT Mi Band warning, which I have no idea why & how I would reboot the band).

I give up and turn on "Write Log Files" option and here's the log that I have:
[gadgetbridge.gz.pdf](https://github.com/Freeyourgadget/Gadgetbridge/files/46365/gadgetbridge.gz.pdf)
I'm not sure since when it's recorded because the log is about 3 MB which is why I have to gzip it, and because github doesn't accept .gz extension I append .pdf to it, so please remove that .pdf before extracting. Hope that's enough to help me figure out what I'm supposed to do.

**Comment by danielegobbetti** [comment]
Hi,

sorry to read you're having problems with the fw install functionality. You read it right that fw series 1.0.10.* are expected to work.

I am from mobile at the moment and can't check your log. what I suggest is that you uninstall any other app that may trigger a transfer of data to the band while you perform the upgrade. This includes the official app by xiaomi and other apps you may have installed (eg for notifications).

let us know if this helps!
--

**Comment by leledumbo** [comment]
no need to rush, I'll patiently wait. The band is now unpaired from Mi Fit, so I don't think there would be any data transfer between them. Well I've tried to uninstall Mi Fit as well, but it remains the same. I don't install any other app that can communicate with the band.

**Comment by cpfeiffer** [comment]
Unfortunately there's not much information about why the update fails. After the first failure, we get some more error codes indicating different "authentication failures". I've never seen something like this before and it may be that these error codes are not the cause, but the effect of the failed update.

So for now, all I can suggest is trying some things out, like
disabling and enabling bluetooth or rebooting the phone and then trying the update again. Or try flashing a different firmware, maybe even the one that you're currently running.

It could also be that Xiaomi added some firmware protection that prevents downgrades or "unauthenticated" updates.
-

**Comment by leledumbo** [comment]
You're right, even flashing the current firmware fails! I wonder what kind of authentication they use for flashing...

**Comment by danielegobbetti** [comment]
@leledumbo, @cpfeiffer  downgrading works for me from 1.0.11.6, so I don't think there's something new to this end (fortunately for us!).

The firmware metadata depend upon the mac address of the band, it may be that for your particular band the checksum that we calculate is wrong because of some corner case that our algorithm is not able to sort out correctly.

Another option (but I tend to rule this out) is that on xiaomi phones there are some kind of "optimizations" while talking to xiaomi peripherals that interfere with third party apps.

**Comment by cpfeiffer** [comment]
@danielegobbetti @leledumbo Indeed it could be worthwhile to try the same procedure on a different device. Not sure if it would even pair with another mobile, but is there anything to lose?
It could also be that the band still has the proper firmware in place and restarting it would cause no harm -- we simply don't know (yet).

In the end it might also be just another defect device :-/
